tasks/shared_libraries/database/shared_db.py

```diff
@@ -188,9 +155,65 @@ def get_user_connection(config: Dict[str, str]) -> Engine:
     connection = _create_connection(
         host=config["host"],
         port=config["port"],
-        database=config["database"],
-        username=config["app_user"],
-        password=config["app_user_password"],
+        database=config["user_database"],
+        username=config["user_username"],
+        password=config["user_password"],
     )
 
     return connection
+
+
+# Retry decorator for functions
+def retry_operational_error(max_retries: int = MAX_RETRIES, backoff_in_seconds: int = INITIAL_BACKOFF_IN_SECONDS,
+                            backoff_factor: int = BACKOFF_FACTOR) -> Callable[[Callable[[], RT]], Callable[[], RT]]:
+    """
+    Decorator takes arguments to adjust number of retries and backoff strategy.
+    Args:
+        max_retries (int): number of times to retry in case of failure.
+        backoff_in_seconds (int): Number of seconds to sleep the first time through.
+        backoff_factor (int): Value of the factor used for backoff.
+    """
+
+    def decorator_retry_operational_error(func: Callable[[], RT]) -> Callable[[], RT]:
+        """
+        Main Decorator that takes our function as an argument
+        """
+
+        @functools.wraps(func)  # Use built in for decorators
+        def wrapper_retry_operational_error(*args, **kwargs) -> RT:
+            """
+            Wrapper that performs our extra tasks on the function
+            """
+            # Initialize the retry loop
+            total_retries = 0
+
+            # Enter loop
+            while True:
+                # Try the function and catch the expected error
+                try:
+                    return func(*args, **kwargs)
+                except OperationalError:
+                    if total_retries == max_retries + 1:
```

Comment:
Shouldn't this be `total_retries == max_retries`? I think your retry logic here will allow for a retry to occur more times than configured.

Reply:
yes after double checking, it is retrying more times than expected.